### DISCUSSION TITLE
jsdt-core 2020-12 and on require Java 11 on osgi

### DISCUSSION THREAD

**Discussion by hazendaz** [revelc/jsdt-core#38] [General]
I believe we are starting to rock the boat on what we can do here with jdk 8 to jdk 11 jump.  I'm going to release jsdt-core as it is on master to pick up in latest formatter plugin release but problem we face post this point is that the osgi libraries are jdk 11 required now.  Since we are taking jars here from some source and rebundling, its highly probable we are getting something in that 30mb that is really compiled to java 11 and it probably would be hard for us to detect.  I don't know how much javascript actually changes in eclipse anyways so maybe its not a huge concern and we just sit where we are.  In light of this, I will push up a jdk11 branch only to house what was needed to change until we can further think about this.

Of a personal note, in my day job working over 500 repos, we have roughly 85% of them all building only with jdk 11 regardless of their runtime but I'm not convienced yet doing that jump would be good for us here and we need to discuss that more on the formatter itself.

**Comment by hazendaz** [comment]
Sorry probably should have opened a discussion - forgot about that :)  This is more of a discussion than any issue.

**Reply by ctubbsii** [reply]
I converted it to a discussion. Apparently, that's a feature. :smiley_cat:

**Comment by ctubbsii** [comment]
I'd be totally fine switching to JDK11 as the minimum requirement. I don't know enough about OSGi to have an opinion on how this affects things, though. My opinion would be to go ahead and push those changes to the `main` branch and bump the major version on this lib.

**Reply by hazendaz** [reply]
ok - for this bit I think that makes most sense.  I don't know osgi enough either and without downstream ci doing 8 and 11, it would be risky trying to stay at 8.  I did start a thread over on formatter plugin as we need to make a decision there that is far more impactful.  No one even noticed this was way out of date so I doubt many would miss it being behind if we keep formatter on jdk 8.  We could carry both but figure there maybe we get some community comments on going 11.  Anyway, think this is good for here.  I'll push the jdk 11 version here out this coming weekend and it can simply sit until we make further decisions (and actually thinking more about that we could just show how to exchange it out while staying jdk 8).  The trade-off there is if you want javascript piece fully up to date it would be 11 with an override.  But otherwise we stay 8 on the core formatter.  Anyway, stuff to think about.

**Reply by ctubbsii** [reply]
I'd be fine switching the main formatter plugin to 11 also, maybe do a major version bump, just to reset the palette, but can discuss over there. I think your plan for this lib is fine.
